src/Params/PADnoteParameters.cpp

```diff
@@ -87,13 +87,14 @@ static const rtosc::Ports realtime_ports =
     {"sample#64:ifb", rProp(internal) rDoc("Nothing to see here"), 0,
         [](const char *m, rtosc::RtData &d)
         {
+            // takes Sample struct for Sample number "n" (see below for "n")
             // MiddleWare calls this to send the generated sample buffers to us
             PADnoteParameters *p = (PADnoteParameters*)d.obj;
             const char *mm = m;
             while(!isdigit(*mm))++mm;
             int n = atoi(mm);
-            p->sample[n].size     = rtosc_argument(m,0).i;
-            p->sample[n].basefreq = rtosc_argument(m,1).f;
+            p->setCurSampleSize(n, rtosc_argument(m,0).i);
+            p->setCurBaseFreq(n, rtosc_argument(m,1).f);
```

Comment:
The whole setting a sample operation is (in my mind at least) a single atomic action. As such do you think it would make sense to have a single setter, which would keep the update somewhat more atomic (not atomic in the multi-threading sense though).